backend/compact-connect/lambdas/nodejs/ingest-event-reporter/lib/event-client.ts

```diff
@@ -0,0 +1,164 @@
+import { Logger } from '@aws-lambda-powertools/logger';
+import { DynamoDBClient, QueryCommand } from '@aws-sdk/client-dynamodb';
+import { unmarshall } from '@aws-sdk/util-dynamodb';
+import { EnvironmentVariablesService } from './environment-variables-service';
+import { IIngestFailureEventRecord, IValidationErrorEventRecord } from './models';
+
+const environmentVariables = new EnvironmentVariablesService();
+
+
+interface EventClientProps {
+    logger: Logger;
+    dynamoDBClient: DynamoDBClient;
+}
+
+
+/*
+ * Event Client that can retrieve validation error and ingest failure events from the License Events
+ * DynamoDB table.
+ */
+export class EventClient {
+    private readonly logger: Logger;
+    private readonly dynamoDBClient: DynamoDBClient;
+
+    public constructor(props: EventClientProps) {
+        this.logger = props.logger;
+        this.dynamoDBClient = props.dynamoDBClient;
+    }
+
+    /*
+     * Returns timestamps for the beginning and end of the previous UTC day
+     */
+    public getYesterdayTimestamps() {
+        const today: Date = new Date();
+        const yesterday: Date = new Date();
+
+        today.setUTCHours(0, 0, 0, 0);
+        yesterday.setUTCHours(0, 0, 0, 0);
+
+        yesterday.setUTCDate(today.getUTCDate() - 1);
+        // Uncomment to manually force today's events into the time window (for development/testing)
+        // today.setUTCDate(today.getUTCDate() + 1);
+
+        return [
+            Number.parseInt(
+                (yesterday.valueOf()/1000).toString()
+            ),
+            Number.parseInt(
+                (today.valueOf()/1000).toString()
+            )
+        ];
+    }
+
+    /*
+     * Returns timestamps for the beginning and end of the previous UTC week (rolling)
+     */
+    public getLastWeekTimestamps() {
+        const today: Date = new Date();
+        const lastWeek: Date = new Date();
+
+        today.setUTCHours(0, 0, 0, 0);
+        lastWeek.setUTCHours(0, 0, 0, 0);
+
+        lastWeek.setUTCDate(today.getUTCDate() - 7);
+        // Uncomment to manually force today's events into the time window (for development/testing)
+        // today.setUTCDate(today.getUTCDate() + 1);
+
+        return [
+            Number.parseInt(
+                (lastWeek.valueOf()/1000).toString()
+            ),
+            Number.parseInt(
+                (today.valueOf()/1000).toString()
+            )
+        ];
+    }
+
+    /*
+     * Queries the data event table for validation errors by looking for each comp
+     */
+    public async getValidationErrors(
+        compact: string,
+        jurisdiction: string,
+        startTimeStamp: number,
+        endTimeStamp: number
+    ): Promise<IValidationErrorEventRecord[]> {
+        this.logger.info('Getting validation errors', {
+            compact: compact,
+            jurisdiction: jurisdiction,
+            start_time_stamp: startTimeStamp,
+            end_time_stamp: endTimeStamp,
```

Comment:
Since these can all be aggregated together with the python lambda logs via CloudWatch Insights, I think it makes sense to stick with the existing log field convention, which is snake_case.